diff -Naur usbutils-0.82/update-usbids.sh.in usbutils-0.82_mv/update-usbids.sh.in
--- usbutils-0.82/update-usbids.sh.in	2009-04-30 16:47:26.000000000 +0100
+++ usbutils-0.82_mv/update-usbids.sh.in	2015-07-16 09:57:45.784146945 +0100
@@ -38,8 +38,6 @@
 
 if [ -f $DEST ] ; then
 	mv $DEST $DEST.old
-	# --reference is supported only by chmod from GNU file, so let's ignore any errors
-	chmod -f --reference=$DEST.old $DEST.new 2>/dev/null || true
 fi
 mv $DEST.new $DEST
 
