diff -Naur json-c-0.10/config.sub json-c-0.10_mod/config.sub
--- json-c-0.10/config.sub	2012-05-31 03:31:35.000000000 +0100
+++ json-c-0.10_mod/config.sub	2014-11-14 09:02:25.975693202 +0000
@@ -249,7 +249,7 @@
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
-	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
+	| arc | arm | armv7 | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
 	| bfin \
 	| c4x | clipper \
 	| d10v | d30v | dlx | dsp16xx \
