diff -Naur udev-164/rules/rules.d/50-udev-default.rules udev-164_mod/rules/rules.d/50-udev-default.rules
--- udev-164/rules/rules.d/50-udev-default.rules	2016-08-02 15:00:06.755953271 +0100
+++ udev-164_mod/rules/rules.d/50-udev-default.rules	2016-08-02 14:58:52.893521088 +0100
@@ -3,14 +3,6 @@
 SUBSYSTEM=="block", SYMLINK{unique}+="block/%M:%m"
 SUBSYSTEM!="block", SYMLINK{unique}+="char/%M:%m"
 
-KERNEL=="zero|full", MODE="0664"
-KERNEL=="random", MODE="0664",  GROUP="system"
-KERNEL=="urandom", MODE="0664",  GROUP="system"
-KERNEL=="null", MODE="0666",  GROUP="system"
-
-# mali device 
-KERNEL=="mali", MODE="0660",  GROUP="NDS_MW"
-
 # input
 KERNEL=="event*",	NAME="input/udev.NO.use.event"
 KERNEL=="ts[0-9]*|uinput",	MODE="0660", GROUP="NDS_MW"
