--- a/configure	2010-09-06 13:52:24.000000000 +0100
+++ b/configure	2013-09-11 11:56:16.538575408 +0100
@@ -4733,7 +4733,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking how to create a ustar tar archive" >&5
 $as_echo_n "checking how to create a ustar tar archive... " >&6; }
 # Loop over all known methods to create a tar archive until one works.
-_am_tools='gnutar plaintar pax cpio none'
+_am_tools='gnutar plaintar cpio none'
 _am_tools=${am_cv_prog_tar_ustar-$_am_tools}
 # Do not fold the above two line into one, because Tru64 sh and
 # Solaris sh will not grok spaces in the rhs of `-'.
@@ -4761,11 +4761,6 @@
     am__tar_='tar chf - "$tardir"'
     am__untar='tar xf -'
     ;;
-  pax)
-    am__tar='pax -L -x ustar -w "$$tardir"'
-    am__tar_='pax -L -x ustar -w "$tardir"'
-    am__untar='pax -r'
-    ;;
   cpio)
     am__tar='find "$$tardir" -print | cpio -o -H ustar -L'
     am__tar_='find "$tardir" -print | cpio -o -H ustar -L'
