diff -Naur libnih-1.0.3/config.sub libnih-1.0.3_Mod/config.sub
--- libnih-1.0.3/config.sub	2010-10-25 03:26:07.000000000 +0100
+++ libnih-1.0.3_Mod/config.sub	2014-11-12 06:47:17.247872380 +0000
@@ -249,7 +249,7 @@
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
-	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
+	| arc | arm | armv7 | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
 	| bfin \
 	| c4x | clipper \
 	| d10v | d30v | dlx | dsp16xx \
