diff -Naurr iptables-1.4.21_orig/extensions/GNUmakefile.in iptables-1.4.21_new/extensions/GNUmakefile.in
--- iptables-1.4.21_orig/extensions/GNUmakefile.in	2013-11-22 11:18:13.000000000 +0000
+++ iptables-1.4.21_new/extensions/GNUmakefile.in	2015-03-18 14:22:33.782101472 +0000
@@ -74,6 +74,10 @@
 	@mkdir -p "${DESTDIR}${xtlibdir}";
 	if test -n "${targets_install}"; then install -pm0755 $^ "${DESTDIR}${xtlibdir}/"; fi;
 
+uninstall: ${targets_install}
+	$(foreach file,$^,$(shell rm -f "${DESTDIR}${xtlibdir}/$(file)"))
+	rm -rf "${DESTDIR}${xtlibdir}"
+
 clean:
 	rm -f *.o *.oo *.so *.a {matches,targets}.man initext.c initext4.c initext6.c;
 	rm -f .*.d .*.dd;
