diff -Naur udev-164/rules/rules.d/50-udev-default.rules udev-164_mod/rules/rules.d/50-udev-default.rules
--- udev-164/rules/rules.d/50-udev-default.rules	2016-04-26 11:10:45.530153680 +0100
+++ udev-164_mod/rules/rules.d/50-udev-default.rules	2016-04-26 11:15:28.194160938 +0100
@@ -3,9 +3,10 @@
 SUBSYSTEM=="block", SYMLINK{unique}+="block/%M:%m"
 SUBSYSTEM!="block", SYMLINK{unique}+="char/%M:%m"
 
-# mem
-KERNEL=="zero|full|random", MODE="0666"
-KERNEL=="null|urandom", MODE="0666",  GROUP="system"
+KERNEL=="zero|full", MODE="0664"
+KERNEL=="random", MODE="0664",  GROUP="system"
+KERNEL=="urandom", MODE="0664",  GROUP="system"
+KERNEL=="null", MODE="0666",  GROUP="system"
 
 # mali device 
 KERNEL=="mali", MODE="0660",  GROUP="NDS_MW"
@@ -13,7 +13,7 @@
 # input
 KERNEL=="event*",	NAME="input/udev.NO.use.event"
 KERNEL=="ts[0-9]*|uinput",	MODE="0660", GROUP="NDS_MW"
-KERNEL=="hidraw*", SUBSYSTEM=="hidraw", MODE="0664", GROUP="RCU_DRI"
+KERNEL=="hidraw*", SUBSYSTEM=="hidraw", MODE="0660", GROUP="RCU_DRI"
 
 # libusb device nodes
 SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", MODE="0660", GROUP="NDS_MW"
@@ -22,20 +22,6 @@
 # block
 SUBSYSTEM=="block", GROUP="NDS_MW"
 
-
-# block-related
-KERNEL=="sch[0-9]*", GROUP="NDS_MW"
-SUBSYSTEM=="scsi_generic", SUBSYSTEMS=="scsi", ATTRS{type}=="0", GROUP="NDS_MW"
-KERNEL=="pg[0-9]*", GROUP="NDS_MW"
-KERNEL=="qft[0-9]*|nqft[0-9]*|zqft[0-9]*|nzqft[0-9]*|rawqft[0-9]*|nrawqft[0-9]*", GROUP="NDS_MW"
-KERNEL=="rawctl", GROUP="NDS_MW"
-SUBSYSTEM=="raw", KERNEL=="raw[0-9]*", GROUP="NDS_MW"
-SUBSYSTEM=="aoe", GROUP="NDS_MW", MODE="0220"
-SUBSYSTEM=="aoe", KERNEL=="err", MODE="0440"
-
 # network
-KERNEL=="tun",			MODE="0666", OPTIONS+="static_node=net/tun"
-
-# CPU
-KERNEL=="cpu[0-9]*",		MODE="0444"
+KERNEL=="tun",			MODE="0660", GROUP="airties", OPTIONS+="static_node=net/tun"
 
