diff -Naur udev-164/rules/rules.d/50-udev-default.rules udev-164_mod/rules/rules.d/50-udev-default.rules
--- udev-164/rules/rules.d/50-udev-default.rules	2015-06-25 14:55:55.106029622 +0100
+++ udev-164_mod/rules/rules.d/50-udev-default.rules	2015-06-25 14:57:13.442031633 +0100
@@ -17,7 +17,8 @@
 KERNEL=="vport*", ATTR{name}=="?*", SYMLINK+="virtio-ports/$attr{name}"
 
 # mem
-KERNEL=="null|zero|full|random|urandom", MODE="0666"
+KERNEL=="zero|full|random", MODE="0666"
+KERNEL=="null|urandom", MODE="0660",  GROUP="system"
 KERNEL=="mem|kmem|port|nvram",	GROUP="root", MODE="0640"
 
 # input
