--- ltp-full-20070228/testcases/network/tcp_cmds/ping/ping01.orig	2007-03-01 04:40:29.000000000 +0000
+++ ltp-full-20070228/testcases/network/tcp_cmds/ping/ping01	2009-07-23 09:57:05.000000000 +0100
@@ -58,7 +58,7 @@
        which $cmd 2>&1 1>/dev/null
 	   if [ $? -ne 0 ]
 	   then
-	       ./tst_resm TBROK "Test broke: command $cmd not found"
+	       tst_resm TBROK "Test broke: command $cmd not found"
 		   exit 1
        fi
    done
@@ -75,11 +75,11 @@
 do_test() 
 {
    $trace_logic
-   ./tst_resm TINFO "ping with 8 16 32 64 128 256 512 1024 2048 4064 ICMP packets"
+   tst_resm TINFO "ping with 8 16 32 64 128 256 512 1024 2048 4064 ICMP packets"
    for packetsize in $PACKETSIZES
    do
-	  ./tst_resm TINFO "calling ping with packet size = $packetsize"
-      ping  -R -c $COUNT -s $packetsize $RHOST  2>&1 1>/dev/null
+	  tst_resm TINFO "calling ping with packet size = $packetsize"
+      ping -c $COUNT -s $packetsize $RHOST  2>&1 1>/dev/null
       [ $? -eq 0 ] || end_testcase "failed: ping -c $COUNT -s $packetsize $RHOST"
       sleep $SLEEPTIME
    done
@@ -99,8 +99,8 @@
 {
    $trace_logic
 
-   [ $# -eq 0 ] && { ./tst_resm TPASS "Test Successful"; exit 0; }
-   ./tst_resm TFAIL "Test Failed: $@"
+   [ $# -eq 0 ] && { tst_resm TPASS "Test Successful"; exit 0; }
+   tst_resm TFAIL "Test Failed: $@"
    exit 1
 }
 
