--- a/tools/bccmd.c	2012-06-13 16:04:20.000000000 +0100
+++ b/tools/bccmd.c	2014-12-09 11:16:35.963694296 +0000
@@ -462,6 +462,26 @@
 	return 0;
 }
 
+static int cmd_e2set(int transport, int argc, char *argv[])
+{
+	uint8_t array[8];
+	uint16_t log2bytes, addmask;
+	OPT_HELP(2, NULL);
+	if (strncasecmp(argv[0], "0x", 2)) {
+		log2bytes = atoi(argv[0]);
+	} else
+		log2bytes = strtol(argv[0] + 2, NULL, 16);
+	if (strncasecmp(argv[1], "0x", 2)) {
+		addmask = atoi(argv[1]);
+	} else
+		addmask = strtol(argv[1] + 2, NULL, 16);
+	memset(array, 0, sizeof(array));
+	array[0] = log2bytes & 0xff;
+	array[1] = log2bytes >> 8;
+	array[2] = addmask & 0xff;
+	array[3] = addmask >> 8;
+	return transport_write(transport, 0x300e, array, 8);
+}
 static int cmd_coldreset(int transport, int argc, char *argv[])
 {
 	return transport_write(transport, CSR_VARID_COLD_RESET, NULL, 0);
@@ -1056,6 +1076,7 @@
 	char *doc;
 } commands[] = {
 	{ "builddef",  cmd_builddef,  "",                    "Get build definitions"          },
+	{ "e2set",     cmd_e2set,     "<log2bytes> <mask>",  "Initialise eeprom"              },
 	{ "keylen",    cmd_keylen,    "<handle>",            "Get current crypt key length"   },
 	{ "clock",     cmd_clock,     "",                    "Get local Bluetooth clock"      },
 	{ "rand",      cmd_rand,      "",                    "Get random number"              },
