--- a/Makefile.in	2014-12-10 11:37:32.458774770 +0000
+++ b/Makefile.in	2014-12-10 11:52:23.678797648 +0000
@@ -136,7 +136,7 @@
 @HID2HCI_TRUE@udev_PROGRAMS = tools/hid2hci$(EXEEXT)
 @HID2HCI_TRUE@am__append_21 = tools/hid2hci.1
 @HID2HCI_FALSE@am__append_22 = tools/hid2hci.1
-@EXPERIMENTAL_TRUE@am__append_23 = tools/bdaddr.1
+@TOOLS_TRUE@am__append_23 = tools/bdaddr.1
 @READLINE_TRUE@am__append_24 = attrib/gatttool \
 @READLINE_TRUE@			tools/obex-client-tool tools/obex-server-tool
 
@@ -260,12 +260,12 @@
 @TOOLS_TRUE@	tools/hcidump$(EXEEXT) tools/rfcomm$(EXEEXT) \
 @TOOLS_TRUE@	tools/rctest$(EXEEXT) tools/l2test$(EXEEXT) \
 @TOOLS_TRUE@	tools/l2ping$(EXEEXT) tools/sdptool$(EXEEXT) \
-@TOOLS_TRUE@	tools/ciptool$(EXEEXT) tools/bccmd$(EXEEXT)
+@TOOLS_TRUE@	tools/ciptool$(EXEEXT) tools/bccmd$(EXEEXT) tools/bdaddr$(EXEEXT)
 @EXPERIMENTAL_TRUE@am__EXEEXT_4 = emulator/btvirt$(EXEEXT) \
 @EXPERIMENTAL_TRUE@	emulator/b1ee$(EXEEXT) \
 @EXPERIMENTAL_TRUE@	tools/mgmt-tester$(EXEEXT) \
 @EXPERIMENTAL_TRUE@	tools/gap-tester$(EXEEXT) \
-@EXPERIMENTAL_TRUE@	tools/bdaddr$(EXEEXT) tools/avinfo$(EXEEXT) \
+@XPERIMENTAL_TRUE@	tools/bdaddr$(EXEEXT) tools/avinfo$(EXEEXT) \
 @EXPERIMENTAL_TRUE@	tools/avtest$(EXEEXT) \
 @EXPERIMENTAL_TRUE@	tools/scotest$(EXEEXT) \
 @EXPERIMENTAL_TRUE@	tools/hcieventmask$(EXEEXT) \
@@ -640,11 +640,11 @@
 @TOOLS_TRUE@tools_bccmd_DEPENDENCIES = lib/libbluetooth-private.la \
 @TOOLS_TRUE@	$(am__DEPENDENCIES_1)
 am__tools_bdaddr_SOURCES_DIST = tools/bdaddr.c src/oui.h src/oui.c
-@EXPERIMENTAL_TRUE@am_tools_bdaddr_OBJECTS = tools/bdaddr.$(OBJEXT) \
-@EXPERIMENTAL_TRUE@	src/oui.$(OBJEXT)
+@TOOLS_TRUE@am_tools_bdaddr_OBJECTS = tools/bdaddr.$(OBJEXT) \
+@TOOLS_TRUE@	src/oui.$(OBJEXT)
 tools_bdaddr_OBJECTS = $(am_tools_bdaddr_OBJECTS)
-@EXPERIMENTAL_TRUE@tools_bdaddr_DEPENDENCIES =  \
-@EXPERIMENTAL_TRUE@	lib/libbluetooth-private.la
+@TOOLS_TRUE@tools_bdaddr_DEPENDENCIES =  \
+@TOOLS_TRUE@	lib/libbluetooth-private.la
 tools_btattach_SOURCES = tools/btattach.c
 tools_btattach_OBJECTS = tools/btattach.$(OBJEXT)
 tools_btattach_LDADD = $(LDADD)
@@ -1423,8 +1423,8 @@
 @TOOLS_TRUE@	$(am__append_18)
 @HID2HCI_TRUE@udevdir = @UDEV_DIR@
 @HID2HCI_TRUE@tools_hid2hci_LDADD = @USB_LIBS@ @UDEV_LIBS@
-@EXPERIMENTAL_TRUE@tools_bdaddr_SOURCES = tools/bdaddr.c src/oui.h src/oui.c
-@EXPERIMENTAL_TRUE@tools_bdaddr_LDADD = lib/libbluetooth-private.la @UDEV_LIBS@
+@TOOLS_TRUE@tools_bdaddr_SOURCES = tools/bdaddr.c src/oui.h src/oui.c
+@TOOLS_TRUE@tools_bdaddr_LDADD = lib/libbluetooth-private.la @UDEV_LIBS@
 @EXPERIMENTAL_TRUE@tools_avinfo_LDADD = lib/libbluetooth-private.la
 @EXPERIMENTAL_TRUE@tools_avtest_LDADD = lib/libbluetooth-private.la
 @EXPERIMENTAL_TRUE@tools_scotest_LDADD = lib/libbluetooth-private.la

--- a/Makefile.tools	2014-12-10 11:14:52.846739868 +0000
+++ b/Makefile.tools	2014-12-10 11:23:14.418752744 +0000
@@ -59,7 +59,7 @@
 if TOOLS
 bin_PROGRAMS += tools/hciattach tools/hciconfig tools/hcitool tools/hcidump \
 			tools/rfcomm tools/rctest tools/l2test tools/l2ping \
-			tools/sdptool tools/ciptool tools/bccmd
+			tools/sdptool tools/ciptool tools/bccmd tools/bdaddr
 
 tools_hciattach_SOURCES = tools/hciattach.c tools/hciattach.h \
 						tools/hciattach_st.c \
@@ -126,6 +126,9 @@
 tools_bccmd_LDADD += @USB_LIBS@
 endif
 
+tools_bdaddr_SOURCES = tools/bdaddr.c src/oui.h src/oui.c
+tools_bdaddr_LDADD = lib/libbluetooth-private.la @UDEV_LIBS@
+
 dist_man_MANS += tools/hciattach.1 tools/hciconfig.1 \
 			tools/hcitool.1 tools/hcidump.1 \
 			tools/rfcomm.1 tools/rctest.1 tools/l2ping.1 \
@@ -134,7 +137,8 @@
 EXTRA_DIST += tools/hciattach.1 tools/hciconfig.1 \
 			tools/hcitool.1 tools/hcidump.1 \
 			tools/rfcomm.1 tools/rctest.1 tools/l2ping.1 \
-			tools/sdptool.1 tools/ciptool.1 tools/bccmd.1
+			tools/sdptool.1 tools/ciptool.1 tools/bccmd.1 \
+			tools/bdaddr.1
 endif
 
 if HID2HCI
