--- busybox-1.13.1-orig/Makefile	2008-11-29 06:49:52.000000000 +0000
+++ busybox-1.13.1/Makefile	2010-02-19 10:40:18.000000000 +0000
@@ -314,7 +314,7 @@
 
 # Use LINUXINCLUDE when you must reference the include/ directory.
 # Needed to be compatible with the O= option
-CFLAGS		:= $(CFLAGS)
+CFLAGS		:= $(CFLAGS) $(EXTRA_CFLAGS)
 # Added only to final link stage of busybox binary
 CFLAGS_busybox	:= $(CFLAGS_busybox)
 CPPFLAGS	:= $(CPPFLAGS)
