diff -rupN tr69-data-model-orig/config.sub tr69-data-model/config.sub
--- tr69-data-model-orig/config.sub	2014-02-18 11:41:33.831771495 +0000
+++ tr69-data-model/config.sub	2014-02-18 11:41:55.930448990 +0000
@@ -248,7 +248,7 @@ case $basic_machine in
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
-	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
+	| arc | arm | armv7 | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
 	| bfin \
 	| c4x | clipper \
 	| d10v | d30v | dlx | dsp16xx \
