diff -Naurr diagnostics-1.3_orig/apps/cmd/tr069.sh diagnostics-1.3_new/apps/cmd/tr069.sh
--- diagnostics-1.3_orig/apps/cmd/tr069.sh	2016-01-11 12:02:17.000000000 +0000
+++ diagnostics-1.3_new/apps/cmd/tr069.sh	2016-01-13 10:18:21.797673919 +0000
@@ -1,9 +1,7 @@
 #!/bin/sh
-PACEROOT=/opt/pace
+PACEROOT=/tr069
 PACEBIN=${PACEROOT}/bin
-PACELIB=${PACEROOT}/lib
-CMDBDIR=/opt/cm
-export LD_LIBRARY_PATH=${PACELIB}
+CMDBDIR=/tr069/cm
 PATH=${PATH}:${PACEBIN}
 if test -f ${PACEBIN}/iptest && test "$1" == "start"; then
 ${PACEBIN}/iptest > /dev/null 2>&1
diff -Naurr diagnostics-1.3_orig/apps/cmfilechk/main.c diagnostics-1.3_new/apps/cmfilechk/main.c
--- diagnostics-1.3_orig/apps/cmfilechk/main.c	2016-01-11 12:02:17.000000000 +0000
+++ diagnostics-1.3_new/apps/cmfilechk/main.c	2016-01-11 13:24:58.489027428 +0000
@@ -12,7 +12,7 @@
 */
 /*
  * Now that TR069 has to be run as a non-root user, there will be some
- * STBs where the files in /opt/cm/ are owned by root:root.
+ * STBs where the files in /tr069/cm/ are owned by root:root.
  * This program will change them to tr069:tr069, before starting the TR069 Daemons.
  */
 
@@ -132,8 +132,8 @@
 int main()
 {
 int r;
-	r = walk_dir("/opt/cm/", "cmlegacy.??", WS_DEFAULT);
-	r = walk_dir("/opt/cm/", "sky_tr69_*", WS_DEFAULT);
-	r = walk_dir("/opt/cm/", "serialnumber.txt", WS_DEFAULT);
+	r = walk_dir("/tr069/cm/", "cmlegacy.??", WS_DEFAULT);
+	r = walk_dir("/tr069/cm/", "sky_tr69_*", WS_DEFAULT);
+	r = walk_dir("/tr069/cm/", "serialnumber.txt", WS_DEFAULT);
 	return 0;
 }
diff -Naurr diagnostics-1.3_orig/apps/fixmount/main.c diagnostics-1.3_new/apps/fixmount/main.c
--- diagnostics-1.3_orig/apps/fixmount/main.c	2016-01-11 12:02:17.000000000 +0000
+++ diagnostics-1.3_new/apps/fixmount/main.c	2016-01-11 13:31:41.896630703 +0000
@@ -1,17 +1,24 @@
 /* Program to fix TR069 Mount Point */
 /* David Roberts v1.0 24/07/2015 */
+/* Initial Version. */
+/* David Roberts v1.1 11/01/2016 */
+/* Also do chown to mount point */
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
 #include <sys/stat.h>
 
+#define SKY_TR069_UID (21000)
+#define SKY_TR069_GID (21000)
+#define FILENAME_BUFFER (256)
+
 int main(int argc, char **argv)
 {
     char mode[] = "0777";
 	/* Must start with this path otherwise fail. */
 	char testpath[] = "/mnt/nds/";
-    char buf[256];
+    char buf[FILENAME_BUFFER];
     int i;
 	memset(buf, 0, sizeof(buf));
 	if(NULL == argv[1]){
@@ -33,6 +40,13 @@
                 argv[0], buf, strerror(errno)); */
         exit(0);
     }
+	/* Change Owner and group to tr069 */
+    if (chown(buf, SKY_TR069_UID, SKY_TR069_GID) < 0){
+		/* Don't give any indication of what this program does. */
+		/*fprintf(stderr, "%s: error in %s, (%s)\n",
+				argv[0], buf, strerror(errno)); */
+		exit(0);
+	}
     return(0);
 }
 
diff -Naurr diagnostics-1.3_orig/apps/mifd/mgmtif/vendor_hook_deviceinfo.c diagnostics-1.3_new/apps/mifd/mgmtif/vendor_hook_deviceinfo.c
--- diagnostics-1.3_orig/apps/mifd/mgmtif/vendor_hook_deviceinfo.c	2016-01-11 12:02:17.000000000 +0000
+++ diagnostics-1.3_new/apps/mifd/mgmtif/vendor_hook_deviceinfo.c	2016-01-11 13:45:05.195896982 +0000
@@ -58,7 +58,7 @@
 #define BSKYB_DEVICEINFO_HOSTNAME_SIZE	(20)
 #define BSKYB_DEVICEINFO_HOSTNAME_FILENAME				"/etc/hostname"
 #define BSKYB_DEVICEINFO_SOFTWARE_VERSION_FILENAME		"/tmp/VERSION.TXT"
-#define BSKYB_DEVICEINFO_SERIAL_NUMBER_FILENAME         "/opt/cm/serialnumber.txt"
+#define BSKYB_DEVICEINFO_SERIAL_NUMBER_FILENAME         "/tr069/cm/serialnumber.txt"
 
 /* Macro for turning OFF unwanted device info parameters */
 #define SKY_TURN_OFF_DEVICEINFO_PARAMETERS
diff -Naurr diagnostics-1.3_orig/apps/mifd/mgmtif/vendor_hook_temperaturestatus_temperaturesensor.c diagnostics-1.3_new/apps/mifd/mgmtif/vendor_hook_temperaturestatus_temperaturesensor.c
--- diagnostics-1.3_orig/apps/mifd/mgmtif/vendor_hook_temperaturestatus_temperaturesensor.c	2016-01-11 12:02:17.000000000 +0000
+++ diagnostics-1.3_new/apps/mifd/mgmtif/vendor_hook_temperaturestatus_temperaturesensor.c	2016-01-11 13:45:20.764411586 +0000
@@ -50,7 +50,7 @@
 /*---------------------------------------------------------------------------
 ** Defines and Macros
 */
-#define VENDOR_TSENSOR_DB             "/opt/cm/sky_tr69_tsensor_db.xml"
+#define VENDOR_TSENSOR_DB             "/tr069/cm/sky_tr69_tsensor_db.xml"
 #define VENDOR_TSENSOR_REC_PREFIX     "Device.DeviceInfo.TemperatureStatus.TemperatureSensor"
 #define VENDOR_TSENSOR_REC_LASTUPDATE "LastUpdate"
 #define VENDOR_TSENSOR_REC_HIGHALARM  "HighAlarmTime"
diff -Naurr diagnostics-1.3_orig/apps/xmppd/xmpp_backup.c diagnostics-1.3_new/apps/xmppd/xmpp_backup.c
--- diagnostics-1.3_orig/apps/xmppd/xmpp_backup.c	2014-12-04 12:41:40.000000000 +0000
+++ diagnostics-1.3_new/apps/xmppd/xmpp_backup.c	2016-01-11 13:48:24.079779843 +0000
@@ -81,7 +81,7 @@
 		{
 			// Signal the main loop to handle the connection request event
 			/* Here is where we put our code to add a conn req event*/	
-			ok = system("/opt/pace/bin/cwmc addwork connreq");	
+			ok = system("/tr069/bin/cwmc addwork connreq");	
 			//	write(fdPipeConnectionRequest[1],"1",1);
 		}
 		else
@@ -472,7 +472,7 @@
 
 		
 	/* Get the XMPP Domain */
-        fp=popen("/opt/pace/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Domain", "r");
+        fp=popen("/tr069/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Domain", "r");
         ok = 0;
         while ( fgets( line, sizeof line, fp))
         {
@@ -489,7 +489,7 @@
         pclose(fp);
 
 	 /* Get the XMPP Resource */
-        fp=popen("/opt/pace/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Resource", "r");
+        fp=popen("/tr069/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Resource", "r");
         ok = 0;
         while ( fgets( line, sizeof line, fp))
         {
@@ -506,7 +506,7 @@
         pclose(fp);
 
 	/* Get the XMPP Password */
-        fp=popen("/opt/pace/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Password", "r");
+        fp=popen("/tr069/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Password", "r");
         ok = 0;
         while ( fgets( line, sizeof line, fp))
         {
@@ -523,7 +523,7 @@
         pclose(fp);
 
         /* Get the XMPP Secret */
-        fp=popen("/opt/pace/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Secret", "r");
+        fp=popen("/tr069/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Secret", "r");
         ok = 0;
         while ( fgets( line, sizeof line, fp))
         {
@@ -540,7 +540,7 @@
         pclose(fp);
 
         /* Get the XMPP Server */
-        fp=popen("/opt/pace/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.ServerHost", "r");
+        fp=popen("/tr069/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.ServerHost", "r");
         ok = 0;
         while ( fgets( line, sizeof line, fp))
         {
@@ -557,7 +557,7 @@
         pclose(fp);
 
 	/* Get the XMPP Username */
-        fp=popen("/opt/pace/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Username", "r");
+        fp=popen("/tr069/bin/mifc -get dslf Device.ManagementServer.X_00D09E_XMPPConnReq.Username", "r");
         ok = 0;
         while ( fgets( line, sizeof line, fp))
         {
diff -Naurr diagnostics-1.3_orig/apps/xmppd/xmpp.c diagnostics-1.3_new/apps/xmppd/xmpp.c
--- diagnostics-1.3_orig/apps/xmppd/xmpp.c	2014-12-04 12:41:40.000000000 +0000
+++ diagnostics-1.3_new/apps/xmppd/xmpp.c	2016-01-11 13:46:01.434458990 +0000
@@ -90,7 +90,7 @@
 		{
 			// Signal the main loop to handle the connection request event
 			xmpp_send_connreq = 1;
-			//ok = system("/opt/pace/bin/cwmc addwork connreq");	
+			//ok = system("/tr069/bin/cwmc addwork connreq");	
 		}
 		else
 		{
diff -Naurr diagnostics-1.3_orig/libs/libboard/info.c diagnostics-1.3_new/libs/libboard/info.c
--- diagnostics-1.3_orig/libs/libboard/info.c	2016-01-11 12:02:17.000000000 +0000
+++ diagnostics-1.3_new/libs/libboard/info.c	2016-01-11 13:44:50.912051670 +0000
@@ -116,7 +116,7 @@
 	int i;
 	int fd;
 	const char *path[] = {
-		"/opt/cm/serialnumber.txt",
+		"/tr069/cm/serialnumber.txt",
 		NULL,
 	};
 
@@ -265,7 +265,7 @@
 	int i;
 	int fd;
 	const char *path[] = {
-		"/opt/cm/serialnumber.txt",
+		"/tr069/cm/serialnumber.txt",
 		NULL,
 	};
 
