diff -Naurr diagnostics-1.3_orig/apps/mifd/mifd.c diagnostics-1.3_new/apps/mifd/mifd.c
--- diagnostics-1.3_orig/apps/mifd/mifd.c	2015-08-08 21:05:55.000000000 +0100
+++ diagnostics-1.3_new/apps/mifd/mifd.c	2015-08-08 21:28:16.985053207 +0100
@@ -396,6 +396,7 @@
 #ifdef DETECT_DIRTY_TABLES
     if (ctx->table_states) {
         free(ctx->table_states);
+		ctx->table_states = NULL;
     }
 #endif
 
@@ -535,6 +536,7 @@
     }
 
     free (oidstr);
+	oidstr = NULL;
     return EOK;
 
 }
@@ -626,6 +628,7 @@
 
 bail:
     free(oidstr);
+	oidstr = NULL;
     return err;
 }
 
@@ -668,14 +671,18 @@
             syslog(LOG_DEBUG, "mifd_unregister_notification: removing entry for path \"%s\"", mconn->path);
             TAILQ_REMOVE(&mcpn->cco_list, mcn, mcn_ent);
             free(mcn->mif_path);
+			mcn->mif_path = NULL;
             free(mcn);
+			mcn = NULL;
         }
 
         if (! mcpn->cco_list.tqh_first) {
             syslog(LOG_DEBUG, "mifd_unregister_notification: removing entry for OID \"%s\"", oidstr);
             TAILQ_REMOVE(&mconn->paths, mcpn, mcpn_ent);
             free(mcpn->oidstr);
+			mcpn->oidstr = NULL;
             free(mcpn);
+			mcpn = NULL;
 
             syslog(LOG_DEBUG, "mifd_unregister_notification: unregistering CM callback for OID \"%s\"", oidstr);
             err = cm_notify_unregister(ctx->cm_ctx, func, co);
@@ -687,6 +694,7 @@
 
 done:
         free(oidstr);
+		oidstr = NULL;
         return EOK;
 }
 
@@ -793,6 +801,7 @@
     s_snprintf(val, sz, "%s%s", prefix, value);
     map_insert(map, name, (const u_char *)val, sz);
     free(val);
+	val = NULL;
     return;
 }
 
@@ -3047,6 +3056,7 @@
         TAILQ_REMOVE(&nodes, mne, mne_ent);
         sts |= miflib_add_node(tid, nsname, mne->mne_name, NULL);
         free(mne);
+		mne = NULL;
     }
 
 #ifdef DETECT_DIRTY_TABLES
@@ -3203,6 +3213,7 @@
 
 #ifdef DETECT_DIRTY_TABLES
         free(states);
+		states = NULL;
 #endif
 
         ctx->frametype = frametype;
@@ -3275,7 +3286,9 @@
                 if(ctx->flags & MIFD_FLAG_STOP) {
                     /* We're stopping */
                     free(inbound_fd);
+					inbound_fd = NULL;
                     free(outbound_fd);
+					outbound_fd = NULL;
                     return 0;
                 }
                 else {
@@ -3320,7 +3333,9 @@
     }
 
     free(inbound_fd);
+	inbound_fd = NULL;
     free(outbound_fd);
+	outbound_fd = NULL;
 
     return 0;
 }
@@ -3478,6 +3493,7 @@
         mifd_dispatch_fds(ctx, pfd_count, pfd_list);
 
         free(pfd_list);
+		pfd_list = NULL;
 
         if(ctx->flags & MIFD_FLAG_STATS) {
 
@@ -3687,17 +3703,23 @@
     /* free up mconn structure */
     if(mconn->path != NULL) {
         free(mconn->path);
+		mconn->path = NULL;
     }
     TAILQ_FOREACH(mcpn, &mconn->paths, mcpn_ent) {
         TAILQ_FOREACH(mcn, &mcpn->cco_list, mcn_ent) {
-            if(mcn->mif_path)
+            if(mcn->mif_path){
                 free(mcn->mif_path);
+				mcn->mif_path = NULL;
+            }
             mcn->mcn_cco = NULL;
         }
-        if(mcpn->oidstr)
+        if(mcpn->oidstr){
             free(mcpn->oidstr);
+			mcpn->oidstr = NULL;
+        }
     }
     free(mconn);
+	mconn = NULL;
     return;
 }
 
@@ -3763,6 +3785,7 @@
                ar_str != NULL ? ar_str : "unknown");
         if (ar_str != NULL) {
             free(ar_str);
+			ar_str = NULL;
         }
         return err;
     }
diff -Naurr diagnostics-1.3_orig/apps/mifd/mifd_diags.c diagnostics-1.3_new/apps/mifd/mifd_diags.c
--- diagnostics-1.3_orig/apps/mifd/mifd_diags.c	2014-12-04 12:41:40.000000000 +0000
+++ diagnostics-1.3_new/apps/mifd/mifd_diags.c	2015-08-08 21:30:54.458633219 +0100
@@ -92,6 +92,7 @@
 	while ((mddo_ent = TAILQ_FIRST(tailq_oids))) {
 		TAILQ_REMOVE(tailq_oids, mddo_ent, ent);
 		free(mddo_ent);
+		mddo_ent = NULL;
 	}
 
 	return EOK;
@@ -453,6 +454,7 @@
 			cm_db_disconnect(cm_ctx);
 			if (result) {
 				free(result);
+				result = NULL;
 			}
 			/* try to file this */
 			mde->mde_intstate = MDE_INTSTATE_FAILED; 
@@ -470,6 +472,7 @@
 			cm_db_disconnect(cm_ctx);
 			if (result) {
 				free(result);
+				result = NULL;
 			}
 			mde->mde_intstate = MDE_INTSTATE_CANCELLED;
 			pthread_mutex_unlock(&mde->mde_lock);
@@ -570,6 +573,7 @@
 			cm_db_disconnect(cm_ctx);
 			if (result) {
 				free(result);
+				result = NULL;
 			}
 			/* try to file this */
 			mde->mde_intstate = MDE_INTSTATE_FAILED; 
@@ -587,6 +591,7 @@
 			cm_db_disconnect(cm_ctx);
 			if (result) {
 				free(result);
+				result = NULL;
 			}
 			mde->mde_intstate = MDE_INTSTATE_CANCELLED;
 			pthread_mutex_unlock(&mde->mde_lock);
@@ -812,6 +817,7 @@
 		mifd_diags_join_mde(mde);
 	}
 	free(mde);
+	mde = NULL;
 	return;
 }
 
@@ -942,6 +948,7 @@
 					 "oid= %s", oidstr);
 			if (!oidstr) {
 				free(oidstr);
+				oidstr = NULL;
 			}
 			continue; /* not sure that there's anything else to do */
 		}
diff -Naurr diagnostics-1.3_orig/apps/mifd/mifd_stats.c diagnostics-1.3_new/apps/mifd/mifd_stats.c
--- diagnostics-1.3_orig/apps/mifd/mifd_stats.c	2015-08-08 20:49:02.000000000 +0100
+++ diagnostics-1.3_new/apps/mifd/mifd_stats.c	2015-08-08 21:31:18.991525631 +0100
@@ -115,6 +115,7 @@
 
         TAILQ_REMOVE(&ctx->statistics, statptr, statistics);
         free(statptr);
+		statptr = NULL;
     }
 }
 
