diff -Naur pigz-2.3.3/Makefile pigz-2.3.3_mod/Makefile
--- pigz-2.3.3/Makefile	2015-01-20 04:12:31.000000000 +0000
+++ pigz-2.3.3_mod/Makefile	2015-06-25 11:55:43.073752072 +0100
@@ -5,7 +5,7 @@
 # use gcc and gmake on Solaris
 
 pigz: pigz.o yarn.o try.o ${ZOPFLI}deflate.o ${ZOPFLI}blocksplitter.o ${ZOPFLI}tree.o ${ZOPFLI}lz77.o ${ZOPFLI}cache.o ${ZOPFLI}hash.o ${ZOPFLI}util.o ${ZOPFLI}squeeze.o ${ZOPFLI}katajainen.o
-	$(CC) $(LDFLAGS) -o pigz $^ -lpthread -lm
+	$(CC) $(LDFLAGS) -o pigz $^ -lpthread -lm -lz
 	ln -f pigz unpigz
 
 pigz.o: pigz.c yarn.h try.h ${ZOPFLI}deflate.h ${ZOPFLI}util.h
