diff -rupN libatasmart-0.19-orig/build-aux/config.sub libatasmart-0.19/build-aux/config.sub
--- libatasmart-0.19-orig/build-aux/config.sub	2014-02-18 10:49:59.811469014 +0000
+++ libatasmart-0.19/build-aux/config.sub	2014-02-18 10:50:25.854270412 +0000
@@ -250,7 +250,7 @@ case $basic_machine in
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
-	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
+	| arc | arm | armv7 | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
         | be32 | be64 \
 	| bfin \
 	| c4x | clipper \
